cpu/cc2538/periph/uart.c

```diff
@@ -61,26 +61,23 @@ enum {
 #define DIVFRAC_MASK          ( (1 << DIVFRAC_NUM_BITS) - 1 )
 
 /** @brief Indicates if there are bytes available in the UART0 receive FIFO */
-#define uart0_rx_avail() ( UART0->cc2538_uart_fr.FRbits.RXFE == 0 )
+#define uart0_rx_avail() ( uart_config[0].dev->cc2538_uart_fr.FRbits.RXFE == 0 )
 
 /** @brief Indicates if there are bytes available in the UART1 receive FIFO */
-#define uart1_rx_avail() ( UART1->cc2538_uart_fr.FRbits.RXFE == 0 )
+#define uart1_rx_avail() ( uart_config[1].dev->cc2538_uart_fr.FRbits.RXFE == 0 )
 
 /** @brief Read one byte from the UART0 receive FIFO */
-#define uart0_read()     ( UART0->DR )
+#define uart0_read()     ( uart_config[0].dev->DR )
 
 /** @brief Read one byte from the UART1 receive FIFO */
-#define uart1_read()     ( UART1->DR )
+#define uart1_read()     ( uart_config[1].dev->DR )
 
 /*---------------------------------------------------------------------------*/
 
 /**
  * @brief Allocate memory to store the callback functions.
  */
-static uart_isr_ctx_t uart_config[UART_NUMOF];
-
-cc2538_uart_t * const UART0 = (cc2538_uart_t *)0x4000c000;
-cc2538_uart_t * const UART1 = (cc2538_uart_t *)0x4000d000;
+static uart_isr_ctx_t uart_ctx[UART_NUMOF];
 
 /*---------------------------------------------------------------------------*/
```

Comment:
I would prefer to drop those 'separators'. We never really use anything like it in the RIOT code base, so they seem to be out of another world :-)